src/main/java/org/apache/maven/plugin/resources/remote/AbstractProcessRemoteResourcesMojo.java

```diff
@@ -622,9 +630,10 @@ protected boolean copyResourceIfExists(File file, String relFileName, VelocityCo
                 source = templateSource;
             }
 
-            if (source.exists() && !source.equals(file)) {
+            if (source.exists() && !source.equals(outputFile)) {
                 if (source == templateSource) {
-                    try (CachingOutputStream os = new CachingOutputStream(file)) {
+                    getLog().debug("Use project resource '" + source + "' as resource with Velocity");
+                    try (CachingOutputStream os = new CachingOutputStream(outputFile)) {
```

Comment:
You don't need to next the try blocks. Stream, writer, and reader can all be set up in the initial clause in parentheses

Reply:
done